Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/NetworkHelper.java

```diff
@@ -0,0 +1,109 @@
+package org.dolphinemu.dolphinemu.utils;
+
+import android.app.Service;
+import android.content.Context;
+import android.net.ConnectivityManager;
+import android.net.LinkAddress;
+import android.net.RouteInfo;
+import android.os.Build;
+
+import androidx.annotation.RequiresApi;
+
+import org.dolphinemu.dolphinemu.NativeLibrary;
+
+import java.net.Inet4Address;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.List;
+
+public class NetworkHelper
+{
+  private static ConnectivityManager GetConnectivityManager()
+  {
+    Context context = NativeLibrary.getEmulationActivity();
```

Comment:
Is there a reason why you need the `EmulationActivity` context specifically? You should be able to use `DolphinApplication.getAppContext()` otherwise.